src/main/kotlin/app/revanced/patches/music/ad/video/fingerprints/ShowMusicVideoAdsFingerprint.kt

```diff
@@ -0,0 +1,26 @@
+package app.revanced.patches.music.ad.video.fingerprints
+
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.extensions.or
+import app.revanced.patcher.fingerprint.method.annotation.FuzzyPatternScanMethod
+import app.revanced.patcher.fingerprint.method.annotation.MatchingMethod
+import app.revanced.patcher.fingerprint.method.impl.MethodFingerprint
+import app.revanced.patches.music.ad.video.annotations.MusicVideoAdsCompatibility
+import org.jf.dexlib2.AccessFlags
+import org.jf.dexlib2.Opcode
+
+@Name("show-video-ads-method-fingerprint")
+@MatchingMethod(
+    "Laghd;", "d"
+)
+@FuzzyPatternScanMethod(2) // FIXME: Test this threshold and find the best value.
+@MusicVideoAdsCompatibility
+@Version("0.0.1")
+object ShowMusicVideoAdsFingerprint : MethodFingerprint(
+    "V", AccessFlags.PUBLIC or AccessFlags.FINAL, listOf("Z"), listOf(
+        Opcode.IPUT_BOOLEAN,
+        Opcode.INVOKE_VIRTUAL,
+        Opcode.RETURN_VOID
+    )
+)
```

Comment:
Is this really finding the method you need?

Reply:
This is the method that it targets: 
![image](https://user-images.githubusercontent.com/4687488/178934413-5174c987-f0e5-455a-ac3b-36f262a6327a.png)
How should I do it then?

Reply:
Today I can confirm with a higher level of certainty  that it's working 😄

Reply:
Try using jadx and find the code.

Reply:
Thank you but no luck too :(

Reply:
If there is someone who can test this just to make sure I'm not wrong, it would be great. I can even provide the binary that I'm using by PM

Reply:
Maybe it's patching to the wrong location?

Reply:
I will test it once i get back to PC and maybe change some things about fingerprint.

Reply:
![image](https://user-images.githubusercontent.com/94744045/178982856-e7ff9236-c6e2-4d95-be16-2ca21cb59c95.png)
yeah it patches

Reply:
Yeah I know I got it with my latest push using the constructor fingerprint!
But it's crashing that is why I didn't tell earlier about this push. I'll try to fix it still today and let you know.
Thank you for the test!!
![image](https://user-images.githubusercontent.com/4687488/178975426-2ac8767d-4980-46c2-9e45-e26d9a149e57.png)

Reply:
> The reason why it was crashing: It is crashing because you try to call set visibility on the instance object inside register p1 but p1 is a boolean and not an instance of view. Replace the first argument of the invoke call with p0 which equals to the this instance as the method is an instance method.

No need for the setVisibility anymore. The patch became even simpler and more close to what the Vanced team did 😄
Are you able to merge it now?